action/sealedenvelope.go

```diff
@@ -179,6 +197,49 @@ func (sealed *SealedEnvelope) loadProto(pbAct *iotextypes.Action, evmID uint32)
 	return nil
 }
 
+func (sealed *SealedEnvelope) loadEvmTransaction(pbAct *iotextypes.Action, evmID uint32) error {
+	var (
+		tx   = types.Transaction{}
+		core = pbAct.GetCore()
+	)
+	if core == nil {
+		return ErrNilProto
+	}
+	if err := tx.UnmarshalBinary(core.GetEvmTxContainer().GetRaw()); err != nil {
+		return err
+	}
+	if tx.Protected() && tx.ChainId().Uint64() != uint64(evmID) {
+		return errors.Wrapf(ErrInvalidAct, "expect chainID = %d, got %d", evmID, tx.ChainId().Uint64())
+	}
+	_, sig, pubkey, err := ExtractTypeSigPubkey(&tx)
+	if err != nil {
+		return err
+	}
+	if len(sig) != 65 {
+		return errors.Errorf("invalid signature length = %d, expecting 65", len(sig))
+	}
+	elp := envelope{
+		AbstractAction: AbstractAction{
+			version:  version.ProtocolVersion,
+			chainID:  core.GetChainID(),
+			nonce:    tx.Nonce(),
+			gasLimit: tx.Gas(),
+			gasPrice: tx.GasPrice(),
+		},
+		payload: NewEvmTxContainer(core.GetChainID(), evmID, &tx),
+	}
+	// populdate all fields
+	sealed.Envelope = &elp
+	sealed.encoding = iotextypes.Encoding_ETHEREUM_CONTAINER
```

Comment:
after converting to a `sealed`, the temp encoding could be replaced

Reply:
not yet, here it is only the temporary housing of evmtx bytes, the final conversion to actual `sealed` (with actual action inside) is done in workingset